bin/hdfcoinc/pycbc_stat_dtphase2

```diff
@@ -0,0 +1,149 @@
+    ra = uniform(0, 2 * numpy.pi, size=size)
+    dec = numpy.arccos(uniform(-1., 1., size=size)) - numpy.pi/2
+    inc = numpy.arccos(uniform(-1., 1., size=size))
+    pol = uniform(0, 2 * numpy.pi, size=size)
+    ip = numpy.cos(inc)
+    ic = 0.5 * (1.0 + ip * ip)
+
+    twidth = .001  # approximate timing error at lower SNRs
+    pwidth = .25   # approximate phase error at lower SNRs
+    serr = 1.4   #
+    sref = 5.0     # Reference SNR for bin smoothing
+    swidth = serr / sref
+
+    # calculate the toa, poa, and amplitude of each sample
+    data = {}
+    for r, ifo in enumerate(args.ifos):
+        data[ifo] = {}
+        fp, fc = d[ifo].antenna_pattern(ra, dec, pol, 0)
+        sp, sc = fp * ip, fc * ic
+        data[ifo]['s'] = ((sp)**2.0 + (sc)**2.0) ** 0.5 * args.relative_sensitivities[r]
+        data[ifo]['t'] = d[ifo].time_delay_from_earth_center(ra, dec, 0)
+        data[ifo]['p'] = numpy.arctan2(sc, sp)
+
+    # Bin the data
+    bind = []
+    keep = None
+    for i in range(len(args.ifos) -1):
+        ifo0 = args.ifos[0 + i]
+        ifo1 = args.ifos[1 + i]
+        dt = (data[ifo0]['t'] - data[ifo1]['t'])
+        dp = (data[ifo0]['p'] - data[ifo1]['p']) % (numpy.pi * 2.0)
+        sr = (data[ifo1]['s'] / data[args.ifos[0]]['s'])
+        dtbin = (dt / twidth).astype(numpy.int)
+        dpbin = (dp / pwidth).astype(numpy.int)
+        srbin = (sr / swidth).astype(numpy.int)
+        
+        srbmax = int((args.snr_ratio / swidth))
+        srbmin = int((1.0 / args.snr_ratio) / swidth)
+        
+        # We'll only store a limited range of ratios
+        if keep is None:
+            keep = (srbin < srbmax) & srbin > (srbmin)
+        else:
+            keep = keep & (srbin < srbmax) & srbin > (srbmin)
+        bind += [dtbin, dpbin, srbin]
+ 
+    # Calculate and sum the weights for each bin
+    # use first ifo as reference for weights
+    bind = [a[keep] for a in bind]
+    
+    w = data[args.ifos[0]]['s'][keep] ** 3            
+    for i, key in enumerate(zip(*bind)):
+        if key not in weights:
+            weights[key] = 0
+        weights[key] += w[i]
+        
+    ol = l
+    l = len(weights.values())
+    logging.info('%s, %s, %s, %s', l, l - ol, (l - ol) / float(size), l / float(nsamples))
+
+logging.info('applying smoothing')
+# apply smoothing iteratively
+for i, ifo in enumerate(range(len(args.ifos)-1)):
+    weights = smooth_param(weights, twidth, i*3 + 0)
+    weights = smooth_param(weights, pwidth, i*3 + 1)
+    weights = smooth_param(weights, swidth, i*3 + 2)    
+logging.info('smoothing done: %s', len(weights))
+
+# save dict to hdf5 file as key + value array
+f = h5py.File(args.output_file, 'w')
+keys = numpy.array(weights.keys())
+values = numpy.array(weights.values())
+f['keys'] = keys
+f['values'] = values / values.max()
+
+f.attrs['srbmin'] = srbmin
+f.attrs['srbmax'] = srbmax
+f.attrs['twidth'] = twidth
+f.attrs['pwidth'] = pwidth
+f.attrs['swidth'] = swidth
+f.attrs['ifos'] = args.ifos
```

Comment:
Does this need the `f.attrs['stat']` magic string?

Reply:
That has now been added.